be/src/storage/tablet_updates.cpp

```diff
@@ -1728,8 +1728,20 @@ Status TabletUpdates::_do_compaction(std::unique_ptr<CompactionInfo>* pinfo) {
     MergeConfig cfg;
     cfg.chunk_size = config::vector_chunk_size;
     cfg.algorithm = algorithm;
-    RETURN_IF_ERROR(compaction_merge_rowsets(_tablet, info->start_version.major_number(), input_rowsets,
-                                             rowset_writer.get(), cfg, cur_tablet_schema));
+
+    // compaction task maybe failed if tablet is deleted
+    st = compaction_merge_rowsets(_tablet, info->start_version.major_number(), input_rowsets, rowset_writer.get(), cfg,
```

Comment:
When dropping the tablet, the state has been `TABLET_SHUTDOWN`. Why will the ingestion find the tablet?

Reply:
> When dropping the tablet, the state has been `TABLET_SHUTDOWN`. Why will the ingestion find the tablet?

Ingestion task maybe start before drop, and during data ingestion, the tablet could be dropped.

Reply:
If the tablet is get, the shared_ptr should guarantee the table is not reclaimed.
The `TABLET_SHUTDOWN` state is to make it not been dropped actually and wait the ingestion/query finished.

Reply:
> If the tablet is get, the shared_ptr should guarantee the table is not reclaimed. The `TABLET_SHUTDOWN` state is to make it not been dropped actually and wait the ingestion/query finished.

Yes, the tablet point does not release until ingestion finished. But the `tablet_meta` will be deleted during dropping tablet. For the primary key model, we will remove index cache, clear delete vector. So the ingestion/compaction maybe failed because the metadata is invalid.